packages/js/components/src/date-time-picker-control/date-time-picker-control.tsx

```diff
@@ -174,72 +174,118 @@ export const DateTimePickerControl: React.FC< DateTimePickerControlProps > = ( {
 	// - [useDebounce hook loses function calls if the dependency changes](https://github.com/WordPress/gutenberg/issues/35505)
 	// - [useMemoOne and useCallbackOne](https://github.com/alexreardon/use-memo-one)
 
-	const onChangePropFunctionRef = useRef<
+	const onChangeRef = useRef<
 		DateTimePickerControlOnChangeHandler | undefined
 	>();
 	useLayoutEffect( () => {
-		onChangePropFunctionRef.current = onChange;
+		onChangeRef.current = onChange;
 	}, [ onChange ] );
 
-	function inputStringChangeHandlerFunction(
-		newInputString: string,
-		fireOnChange: boolean
-	) {
-		if ( ! isMounted.current ) return;
+	function setInputStringWithMoment( dateTime: Moment ) {
+		const newInputString = dateTime.isValid()
+			? formatMoment( dateTime )
+			: dateTime.creationData().input?.toString() || '';
+
+		setInputString( newInputString );
+	}
+
+	function callOnChange( dateTime: Moment ) {
+		const valueToSend = dateTime.isValid()
```

Comment:
If we go with the above suggestion, we might be able to move this formatting logic into that same method.  Though it might need a second argument for format type (iso):

```
function formatDateTime( dateTime: moment, iso: boolean )
```

Reply:
I'm going to keep things separate for readability (I feel like a param like `iso` is a bit opaque)...

`formatDateTimeForDisplay()` and `formatDateTimeAsISO()`